crates/nu-parser/src/parse_keywords.rs

```diff
@@ -1419,21 +1416,11 @@ pub fn parse_export_in_block(
             // further error will be generated by detail `parse_xxx` function.
             working_set.parse_errors.truncate(starting_error_count);
 
-            let is_help = is_help_call || {
-                let decl = working_set.get_decl(decl_id);
-                check_call(working_set, call_span, &decl.signature(), &call);
-                if let Ok(is_help) = has_flag_const(working_set, &call, "help") {
-                    is_help
-                } else {
-                    return garbage_pipeline(working_set, &lite_command.parts);
-                }
-            };
-
-            if starting_error_count != working_set.parse_errors.len() || is_help {
+            if is_help_call || call_kind != CallKind::Valid {
```

Comment:
Do we still need the local `is_help_call` from above with your new `CallKind` logic or could this be a match on both `CallKind::Valid | CallKind::help`?